client.go

```diff
@@ -241,22 +241,79 @@ func (acc *Account) Login(username, password string) error {
 	}
 	defer resp.Body.Close()
 
+	if resp.StatusCode == http.StatusUnauthorized {
```

Comment:
What we did in the other clients is have the status code handling all in one place. So we just map status to error codes and decode the failures. Is this something that's applicable to Go too? So we just automatically handle all of the error types centrally?

Reply:
we should. I will address this on a different pr.